components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/model/handlers/UsernameValidationConfigurationHandler.java

```diff
@@ -156,4 +160,50 @@ private String getUsernameRegEx(RealmConfiguration realmConfig) {
         return new String();
     }
 
+    /**
+     * Perform post actions after validation configuration are updated.
+     *
+     * @param tenantDomain      Tenant Domain.
+     * @param configuration     Updated validation configuration.
+     */
+    @Override
+    public void postValidationConfigurationUpdateHandler(String tenantDomain, ValidationConfiguration configuration)
+            throws InputValidationMgtServerException {
+
+        for (RulesConfiguration rule: configuration.getRules()) {
+            /*
+             Set required and support by default properties of the email attribute to true, when username field
+             validation configurations are set to emailFormatValidator.
+             */
+            if ("EmailFormatValidator".equals(rule.getValidatorName())) {
+                if (Boolean.parseBoolean(rule.getProperties().get(ENABLE_VALIDATOR))) {
+                    try {
+                        updateEmailClaim(tenantDomain);
+                    } catch (ClaimMetadataException e) {
+                        throw new InputValidationMgtServerException(e);
+                    }
+                }
+                break;
+            }
+        }
+    }
+
+    /**
+     * Set required and support by default properties of the email attribute to true.
+     *
+     * @param tenantDomain   Tenant Domain.
+     */
+    private void updateEmailClaim(String tenantDomain) throws ClaimMetadataException {
+        List<LocalClaim> localClaimList = InputValidationDataHolder.getInstance()
+        .getClaimMetadataManagementService().getLocalClaims(tenantDomain);
```

Comment:
Iterate on this only if localClaimList is not epmty empty

Reply:
Addressed